code/game/objects/structures/mirror.dm

```diff
@@ -8,6 +10,30 @@
 	anchored = 1
 	var/shattered = 0
 
+	/// Visual object for handling the viscontents
+	var/datum/weakref/ref
+	vis_flags = VIS_HIDE
+	var/timerid = null
+
+/obj/structure/mirror/New()
+	var/obj/effect/reflection/reflection = new(src.loc)
+	reflection.setup_visuals(src)
+
+	ref = makeweakref(reflection)
+
+	entered_event.register_event(entered_event, reflection, "check_vampire_enter")
```

Comment:
Does this even work anymore? I think the new 515 syntax to call procs dynamically requires you to do this, but even if it doesn't, you might want to do it like this, so that you're using the compiler to check that the procs exist rather than using strings which are only checked at runtime:
```suggestion
	entered_event.register_event(entered_event, reflection, /obj/effect/reflection/proc/check_vampire_enter)
```

To be honest, I don't see why this is using the event system at all. Since the reflection seems to be doing everything it needs to do without any data or procs belonging to the mirror, shouldn't you be overriding /Entered and /Exited on `/obj/effect/reflection directly?`